frontend/catalyst/utils/runtime.py

```diff
@@ -122,6 +122,14 @@ def validate_device_capabilities(
     if hasattr(device, "operations") and hasattr(device, "observables"):
         # For gates, we require strict match
         device_gates = filter_out_adjoint(set(device.operations))
+        if device_name == "lightning.kokkos":
+            # Lightning-kokkos device supports GlobalPhase via PythonAPI but does not via C++ API.
+            # device.operations represents PythonAPI so we relax the validation for this case.
+            # https://github.com/PennyLaneAI/pennylane-lightning/pull/642#issuecomment-2075771670
+            device_gates = device_gates - {"C(GlobalPhase)"}
```

Comment:
I don't like these special cases to be honest. There is a cost/benefit analysis we have to make here. Ultimately what the Python device states is irrelevant, the toml file is the source of truth for the compiler and it will decompose accordingly.
The verification that the set of toml gates matches the set of python device gates was merely added as a nice to have so that it wouldn't go unnoticed if a new gate is added to the python but not the toml file. One could even argue that we shouldn't have this verification at all since the Python device declarations are not relevant to the compiler.

Now recently this internal verification (for our developers) was made stricter and this is leading to problems like having to hardcode special cases. I would rather lean towards removing or relaxing it rather than working in contortions around the rules.

Reply:
> What do you think about removing the validation simultaneously with removing the old device api?

Alright I could be persuaded of that 😅